hack/multi-cluster-test.sh

```diff
@@ -0,0 +1,57 @@
+#!/bin/bash
+#
+# Copyright 2023 The Primaza Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+TAG=$(git rev-parse --short HEAD)
+
+CONTROLLER=primaza-controller:${TAG}
+AGENTAPP=agentapp:${TAG}
+AGENTSVC=agentsvc:${TAG}
+
+if [[ ! -e bin/yq ]]; then
+    make yq
+fi
+
+kind delete clusters worker controller
+
+echo "creating main cluster" && \
+    kind create cluster -q --name controller --kubeconfig out/controller-kubeconfig && \
+    bin/yq -i ".clusters[0].cluster.server = \"https://$(docker container inspect controller-control-plane | bin/yq '.[0].NetworkSettings.Networks.kind.IPAddress'):6443\"" out/controller-kubeconfig &
+
+echo "creating worker cluster" && \
+    kind create cluster -q --name worker --kubeconfig out/worker-kubeconfig && \
+    bin/yq -i ".clusters[0].cluster.server = \"https://$(docker container inspect worker-control-plane | bin/yq '.[0].NetworkSettings.Networks.kind.IPAddress'):6443\"" out/worker-kubeconfig &
+wait
+
+make primaza docker-build IMG="${CONTROLLER}"
+make agentapp docker-build IMG="${AGENTAPP}"
+make agentsvc docker-build IMG="${AGENTSVC}"
+
+for name in "controller" "worker"; do
+    for image in $CONTROLLER $AGENTAPP $AGENTSVC; do
+        kind load docker-image "${image}" --name "${name}" &
+    done
+done
+wait
```

Comment:
this will wait just for the last one

Reply:
Also tested with this script, `wait` acts as an execution barrier on background jobs.

<details><summary>Details</summary>
<p>

```bash
#!/bin/bash

sleep 2 && echo "job a: slept for 2 seconds" &
sleep 1 && echo "job b: slept for 1 seconds" &

wait

echo "all jobs completed!"
```

</p>
</details>

Reply:
sorry, I mistakenly thought that it waits just for the last one. good to know 😄